### DISCUSSION TITLE
Extensible viewport effect

### DISCUSSION THREAD

**Discussion by chengenzhao** [AlmasB/FXGL#1165] [Ideas]
Hi:

Is there any way to extend effects on viewport?
currently there are only 3 ways of viewport effect: shake, fade and flash which works perfectly nice
and is it possible for developers to create their own viewport effect?
like if the protagonist gets hit the viewport border becomes red and then fade out to warn the player
we read the source code and seems like we have to get the camera first
but camera is the internal variable so it can't be used outside the viewport class
<img width="446" alt="截屏2022-04-09 下午12 23 11" src="https://user-images.githubusercontent.com/5525436/162556126-9b0c40fd-306f-4952-999a-ad394049bafc.png">
is it possible to provide a public method to allow developer to extend the effect then add in the camera view components?

Thanks

**Comment by AlmasB** [comment]
Hi, I don't think this is possible with existing API. It sounds useful and should be feasible to implement, so I'm going to move this to an issue.
